### PR TITLE
Cv 162 implement map

### PR DESCRIPTION
**Pull request by Palm394**
- initiate overview page layout
- setup google map
- marker state

create `.env.local` for google map api secret key
and watch out `.gitignore` when you push commit

### CODE REVIEW

File: src/constants/ROUTE.tsx [hunk 1]
```diff
@@ -10,3 +10,12 @@ export enum ROUTE {
...
 }
+
+export enum ASSETS_PATH {
+	MAP_ASSETS = '/mapAssets',
+	MAP_PIN = MAP_ASSETS + '/PIN',
+	CAR_PIN = MAP_PIN + '/car_pin_',
+	RSU_PIN = MAP_PIN + '/rsu_pin.svg',
+	MAP_CAR_PROFILE = MAP_ASSETS + '/PROFILE/car_profile_',
+	MAP_RSU_PROFILE = MAP_ASSETS + '/PROFILE/rsu_profile.svg'
+}
```
Comment (Kkr1328) [inline src/constants/ROUTE.tsx:21]: What do they for?

File: src/types/OVERVIEW.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    radius: number;
+    recommendSpeed: string;
+}
+
+export interface DriverProps {
+	id: string;
+	first_name: string;
+	last_name: string;
+	phone_no: string;
+}
```
Comment (Kkr1328) [inline src/types/OVERVIEW.ts:39]: As I remember, driver type is exist

File: src/types/OVERVIEW.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { PILL_LABEL } from "@/constants/LABEL";
+import { Position } from "./COMMON";
+
+export interface CameraType {
+	name: string;
+	position: Position;
+}
```
Comment (Kkr1328) [inline src/types/OVERVIEW.ts:7]: As I remember, camera type is exist

File: src/components/overview/RSUMarker.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+export default function RSUMarker(props: RSUMarkerProps) {
+    return (
+        <>
+            <Circle center={props.location} radius={props.radius} options={{ fillColor: "#17A5D3", fillOpacity: 0.1, strokeColor: "#17A5D3" }} />
+            <Marker 
+                icon={{
+                    url: ASSETS_PATH.RSU_PIN,
+                    scaledSize: props.isFocus ? new google.maps.Size(84, 84) : new google.maps.Size(64, 64)
```
Comment (Kkr1328) [inline src/components/overview/RSUMarker.tsx:18]: Could color code and size be constant?

File: src/components/overview/CarCardDetail.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  const [expand, setExpand] = useState<boolean>(false)
+
+  return (
+    <>
+      {/* head */}
+      <div className="text-[20px] font-bold flex items-center gap-16 mb-12">
+        <div className="flex gap-8">
+          <Image 
+            src={`${ASSETS_PATH.MAP_CAR_PROFILE}${car.status}.svg`} 
+            width="32" height="32"
```
Comment (Kkr1328) [inline src/components/overview/CarCardDetail.tsx:57]: Could it be constant?

File: src/components/overview/CarCardDetail.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  )
+}
+
+function InCard(car: CarCard) {
+  const [expand, setExpand] = useState<boolean>(false)
+
+  return (
+    <>
+      {/* head */}
+      <div className="text-[20px] font-bold flex items-center gap-16 mb-12">
```
Comment (Kkr1328) [inline src/components/overview/CarCardDetail.tsx:53]: you can use text variant I create instead

File: src/components/overview/CarCardDetail.tsx [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+      </div>
+
+      {/* properties */}
+      <div>Speed: {car.speed ?? "null"}</div>
+      <Collapse in={expand} timeout="auto">
+          <div className='my-4'>Driver: {`${car.driver.first_name} ${car.driver.last_name}`}</div>
+          <div className='my-4'>Phone No. : {car.driver.phone_no ?? "null"}</div>
+          {car.cameras.map((camera) => (
+            <div className='my-4'>{camera.position + " camera"} : {camera.name}</div>
+          ))}
```
Comment (Kkr1328) [inline src/components/overview/CarCardDetail.tsx:72]: Please add text variant

File: src/constants/ROUTE.tsx (hunk 1 above)
Comment (Palm394) [reply src/constants/ROUTE.tsx:21]: map assets path for map material

File: src/components/overview/RSUMarker.tsx (hunk 4 above)
Comment (Palm394) [reply src/components/overview/RSUMarker.tsx:18]: For color, [reference javascript](https://tailwindcss.com/docs/configuration#referencing-in-java-script) is my approach in head now. It is trade-off between bigger client-side bundle size and convenient for modification. Which one is better?

File: src/constants/ROUTE.tsx (hunk 1 above)
Comment (Kkr1328) [reply src/constants/ROUTE.tsx:21]: Firstly, I intend to use this file for URL, not file path routing. However, if you think it goes the same way and is appropriate, I am also okay with that.

File: src/components/overview/RSUMarker.tsx (hunk 4 above)
Comment (Kkr1328) [reply src/components/overview/RSUMarker.tsx]: I believe that file size isn't a significant concern in this context. Additionally, maintaining a constant value can help prevent errors related to color codes and numbers.

File: src/constants/ROUTE.tsx (hunk 1 above)
Comment (Palm394) [reply src/constants/ROUTE.tsx]: I move to constants/OVERVIEW.ts instead

File: src/components/overview/RSUMarker.tsx (hunk 4 above)
Comment (Palm394) [reply src/components/overview/RSUMarker.tsx]: With typescript problem, I can't do it with ts-ignore. so I just move it to `constant/OVERVIEW.ts` instead. I think it enough for this context

File: src/app/dashboard/overview/page.tsx [hunk 8]
```diff
@@ -1,10 +1,147 @@
...
+		if (value !== null) {
+			setPillMode(value)
+		}
+	}
+	
+	function clickCarCard(carID: string) {
+		let index = MockedCarLocation.findIndex((value) => value.id === carID)
+		let target = MockedCarLocation[index]
+		changeFocus(target)
+	}
```
Comment (ChariU) [inline src/app/dashboard/overview/page.tsx:58]: (Not sure is it this code or not) Can't we deselect the `CarCard` once we selected even though we see the `ALL` tab?  Or is there a reason not to deselect it?

https://github.com/Kkr1328/C-V2X-Frontend/assets/62459518/9995b82c-ca93-4080-9043-c16a64da1271

**Review by Kkr1328 — APPROVED** [review]

File: src/app/dashboard/overview/page.tsx (hunk 8 above)
Comment (Palm394) [reply src/app/dashboard/overview/page.tsx:58]: For deselecting, I have planned to do in future PR (maybe `Implement car info card`). In this pull request focus on google map integration only. 

~~ I haven't designed clearly user flow for this event.

**Review by ChariU — APPROVED** [review]

**Pull request merged**